packages/admin/admin/src/form/ChipSelect.tsx

```diff
@@ -0,0 +1,187 @@
+import { ChevronDown } from "@comet/admin-icons";
+import {
+    Chip as MuiChip,
+    ChipProps,
+    ComponentsOverrides,
+    InputBase as MuiInputBase,
+    InputBaseProps,
+    MenuItem,
+    Select,
+    SelectChangeEvent,
+} from "@mui/material";
+import { css, Theme, useThemeProps } from "@mui/material/styles";
+import * as React from "react";
+
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type ChipSelectClassKey = "root" | "inputBase" | "inputRoot" | "chip" | "select";
+
+type OwnerState = {
+    fullWidth: boolean;
+};
+
+const Root = createComponentSlot("div")<ChipSelectClassKey, OwnerState>({
+    componentName: "ChipSelect",
+    slotName: "root",
+})(
+    ({ ownerState }) => css`
+        ${!ownerState.fullWidth &&
+        css`
+            width: fit-content;
+        `}
+    `,
+);
+
+const ChipInputRoot = createComponentSlot("div")<ChipSelectClassKey>({
+    componentName: "ChipSelect",
+    slotName: "inputRoot",
+})(
+    () => css`
+        display: flex;
+        flex-direction: column;
+        position: relative;
+    `,
+);
+
+const Chip = createComponentSlot(MuiChip)<ChipSelectClassKey>({
+    componentName: "ChipSelect",
+    slotName: "chip",
+})(
+    () =>
+        css`
+            justify-content: flex-start;
+        `,
+);
+
+const InputBase = createComponentSlot(MuiInputBase)<ChipSelectClassKey>({
+    componentName: "ChipSelect",
+    slotName: "inputBase",
+})(
+    () => css`
+        .MuiInputBase-input {
+            padding: 0;
+            height: 100% !important;
+            min-height: unset !important;
+        }
+    `,
+);
+
+export interface ChipSelectProps<T = string>
+    extends ThemedComponentBaseProps<{
+            root: "div";
+            inputRoot: "div";
+            select: typeof Select;
+            menuItem: typeof MenuItem;
+            inputBase: typeof MuiInputBase;
+        }>,
+        Omit<ChipProps, "children" | "onChange"> {
+    children?: React.ReactNode;
+    getOptionLabel?: (option: T) => string;
+    getOptionValue?: (option: T) => string;
+    options?: T[];
+    selectedOption?: T;
+    onChange: (event: SelectChangeEvent) => void;
+    fullWidth?: boolean;
+}
+
+const ChipInput = ({
+    chipProps,
+    inputBaseProps,
+    inputRootProps,
+    ...restProps
+}: InputBaseProps & {
+    chipProps?: Omit<ChipProps, "children">;
+    inputBaseProps?: InputBaseProps;
+    inputRootProps?: React.ComponentPropsWithoutRef<"div">;
+}) => {
+    return (
+        <ChipInputRoot {...inputRootProps}>
+            <Chip icon={<ChevronDown />} label={restProps.value?.toString().length ? restProps.value : ""} variant="filled" {...chipProps} />
```

Comment:
This should show the label, not the value

Reply:
Yes I'm working on it. @jamesricky pointed that out already: https://github.com/vivid-planet/comet/pull/2186#discussion_r1647355411

It's a bit trickier than one might think because of the combination of components, etc.

Reply:
https://github.com/vivid-planet/comet/pull/2186/commits/64067616a48fd9a3b6a88441005707739d1dec0d